packages/legacy/core/App/components/tour/CredentialOfferTourSteps.tsx

```diff
@@ -1,28 +1,28 @@
 import * as React from 'react'
-import { useTranslation } from 'react-i18next'
-import { Text } from 'react-native'
 
-import { useTheme } from '../../contexts/theme'
+import { Text } from 'react-native'
 import { RenderProps, TourStep } from '../../contexts/tour/tour-context'
 
 import { TourBox } from './TourBox'
+import useCommonTourHooks from '../../hooks/modal-tour-props'
+import { DispatchAction } from '../../contexts/reducers/store'
 
 export const credentialOfferTourSteps: TourStep[] = [
   {
     Render: (props: RenderProps) => {
       const { currentTour, currentStep, next, stop, previous } = props
-      const { t } = useTranslation()
-      const { ColorPallet, TextTheme } = useTheme()
+      const { t, ColorPallet, TextTheme, closeModalTour } = useCommonTourHooks()
```

Comment:
I'm wondering if we can find a way to not have to pass new functions as props every time a tour step is rendered, especially since they are all the same format.

Could we possibly pass the relevant callback (`stop` in all these cases) and `DispatchAction` into `useCommonTourHooks` and get an `endTour` function back that handles everything? Rather than a `closeModalTour` function that we have to add to another function.

Something like this:
```suggestion
      const { t, ColorPallet, TextTheme, endTour } = useCommonTourHooks(stop, DispatchAction.UPDATE_SEEN_CREDENTIAL_OFFER_TOUR)
```